src/ImpactEvaluator.sol

```diff
@@ -146,25 +136,13 @@ contract ImpactEvaluator is AccessControl {
         address payable[] memory addresses,
         uint64[] memory scores
     ) private {
-        uint totalAmount = 0;
-        uint[] memory amounts = new uint[](addresses.length);
-
-        for (uint i = 0; i < addresses.length; i++) {
-            uint score = scores[i];
-            uint256 amount = (score * roundReward) / MAX_SCORE;
-            amounts[i] = amount;
-            totalAmount += amount;
-        }
-
-        require(address(this).balance >= totalAmount, "Not enough funds");
```

Comment:
Could you please explain why we no longer need to enforce this check?

I am not questioning your change, just want to understand the reasoning.

Reply:
Maybe we can replace this remove check with new checks that may be cheaper to compute:

1. In this place, validate that `address(this).balance >= roundReward` (we have enough funds to pay reward for one round)
2. In the loop over all rewarded addresses, keep track of `totalAmount` paid so far, and abort the loop if we go over `roundReward`.

Thoughts?

Reply:
moved to Slack https://filecoinproject.slack.com/archives/C03PQG6UT2B/p1698924505273569